conman/routes/managers.py

```diff
@@ -72,3 +73,16 @@ def move_branch(self, old_url, new_url):
             Value(new_url),
             Substr('url', len(old_url) + 1),  # 1 indexed
         ))
+
+    def with_level(self, level=None):
+        """
+        Annotate the queryset with the (1-indexed) level of each item.
+
+        The level reflects the number of forward slashes in the path.
```

Comment:
Do we guarantee elsewhere that a `url` always starts and ends with a `/`?

Reply:
Yes, it's a strange necessity of conman -- it makes other similar operations easy.

Reply:
On the whole, it shouldn't cause too many issues, I hope.